qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1094,6 +1095,77 @@ def categories(self):
 
         return cols
 
+    def update(self, md_template):
+        r"""Update values in the sample template
+
+        Parameters
+        ----------
+        md_template : DataFrame
+            The metadata template file contents indexed by samples Ids
+
+        Raises
+        ------
+        QiitaDBError
+            If md_template and db do not have the same sample ids
+            If md_template and db do not have the same column headers
+            If self.can_be_updated is not True
+        """
+        conn_handler = SQLConnectionHandler()
+
+        # Clean and validate the metadata template given
+        new_map = self._clean_validate_template(md_template, self.study_id,
+                                                self.columns_restrictions)
+        # Retrieving current metadata
+        current_map = self._transform_to_dict(conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.{0} WHERE {1}=%s".format(self._table,
+                                                          self._id_column),
+            (self.id,)))
+        dyn_vals = self._transform_to_dict(conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.{0}".format(self._table_name(self.id))))
+
+        for k in current_map:
+            current_map[k].update(dyn_vals[k])
+            current_map[k].pop('study_id', None)
```

Comment:
Actually, taken a closer look to the code, this can be refactored so it can be highly simplified. Lines 1119 to 1131 can be changed by these ones:

```python
current_map = self._transform_to_dict(conn_handler.execute_fetchall(
    "SELECT * FROM qiita.{0}".format(self._table_name(self.id))))
current_map = pd.DataFrame.from_dict(current_map, orient='index')
```

The reason behind is that all the metadata is stored in the dynamic table, and the query in 1119 is only retrieving the `study_id` which we are then filtering in line 1128. The current code looks like leftover from the previous DB schema that I forgot to change. Sorry about that, but at least I saw it now!